src/synthesizer/csharp/mod.rs

```diff
@@ -0,0 +1,668 @@
+                    indent: INDENT,
+                    leading: Some(match structure {
+                        Structure::Composite(name) => match *name {
+                            "Tag" => "new CfnTag\n{".into(),
+                            name => match root_resource {
+                                Some(r) => format!("new Cfn{r}.{name}Property\n{{").into(),
+                                None => unreachable!("cannot emit ResourceIr::Object without a parent resource type")
+                            }
+                        }
+                        Structure::Simple(cfn) => match cfn {
+                            CfnType::Json => "new Dictionary<string, object>\n{".into(),
+                            _ => unreachable!("cannot emit ResourceIr::Object with non-json simple structure ({:?})", cfn)
+                        }
+                    }),
+                    trailing: Some("}".into()),
+                    trailing_newline: false,
+                });
+
+                for (name, val) in properties {
+                    object_block.text(match structure {
+                        Structure::Composite(_) => format!("{name} = "),
+                        Structure::Simple(_) => format!("{{ \"{name}\", "),
+                    });
+                    match val {
+                        ResourceIr::Bool(_) | ResourceIr::Number(_) | ResourceIr::Double(_) => {
+                            object_block.text("\"");
+                            val.emit_csharp(&object_block, root_resource);
+                            object_block.text("\"");
+                        }
+                        _ => val.emit_csharp(&object_block, root_resource),
+                    }
+                    object_block.text(match structure {
+                        Structure::Composite(_) => ",",
+                        Structure::Simple(_) => "},",
+                    });
+                    object_block.newline();
+                }
+            }
+
+            ResourceIr::If(cond, when_true, when_false) => {
+                output.text(format!("{} ? ", camel_case(cond)));
+                when_true.emit_csharp(output, root_resource);
+                output.text(" : ");
+                when_false.emit_csharp(output, root_resource);
+            }
+            ResourceIr::Join(sep, list) => {
+                let items = output.indent_with_options(IndentOptions {
+                    indent: INDENT,
+                    leading: Some(format!("string.Join(\"{sep}\", new []\n{{").into()),
+                    trailing: Some("})".into()),
+                    trailing_newline: false,
+                });
+                for item in list {
+                    item.emit_csharp(&items, root_resource);
+                    items.text(",");
+                    items.newline();
+                }
+            }
+            ResourceIr::Split(sep, str) => {
+                output.text(format!("Fn.Split(\"{sep}\", "));
```

Comment:
interesting. This is a divergence in other langaguages (e.g. typescript actually just calls split in native, but I guess if it's on a `ref` you would have to add more complexity 🤔

Reply:
Yeah, I cut a corner here and used `Fn.split` in all cases. I can update this to use native c# `.Split()` when possible

Reply:
Ok, fixed!